ssm/ssm2.py

```diff
@@ -311,6 +333,90 @@ def _send_msg(self, message, msgid):
             # If it fails, use the v3 metod signiture
             self._conn.send(to_send, headers=headers)
 
+    def pull_msg_rest(self):
+        """Pull a message via HTTPS from self._dest."""
+        if self._protocol != 'HTTPS':
+            # Then this method should not be called,
+            # raise an exception if it is.
+            raise Ssm2Exception('pull_msg_rest called, '
+                                'but protocol not set to HTTPS. '
+                                'Protocol: %s' % self._protocol)
+
+        if self._dest_type == "AMS":
+            self._pull_msg_rest_ams()
+        else:
+            raise Ssm2Exception('Unsupported Destination Type.'
+                                'Type: %s' % self._dest_type)
+
+    def _pull_msg_rest_ams(self):
+        """Pull 1 message from the AMS and acknowledge it."""
+        # This method is setup so that you could pull down and
+        # acknowledge more than one method at a time, but
+        # currently there is no use case for it.
+        messages_to_pull = 1
+        # ack id's will be stored in this list and then acknowledged
+        ackids = []
+
+        for msg_ack_id, msg in self._ams.pull_sub(self._listen,
+                                                  messages_to_pull):
+            # Get the AMS message id
+            msgid = msg.get_msgid()
+            # Get the SSM dirq id
+            try:
+                empaid = msg.get_attr().get('empaid')
+            except AttributeError:
+                # A message without an empaid could be received if it wasn't
+                # sent via the SSM, we need to pull down that message
+                # to prevent it blocking the message queue.
+                log.debug("Message %s has no empaid.", msgid)
+                empaid = "N/A"
+            # get the message body
+            body = msg.get_data()
+
+            log.info('Received message. ID = %s, Argo ID = %s', empaid, msgid)
+
+            extracted_msg, signer, err_msg = self._handle_msg(body)
```

Comment:
This and bit below duplicated (to 405) from STOMP version. Should factor out to method and just use that. Take care of ack parts for AMS.

Reply:
Deferred to #95.